static/js/lib/ckeditor/plugins/ResourceEmbed.ts

```diff
@@ -14,11 +14,9 @@ import {
   RESOURCE_EMBED,
   RESOURCE_EMBED_COMMAND
 } from "./constants"
-import { Shortcode, makeHtmlString } from "./util"
+import { Shortcode, makeHtmlString, replaceShortcodes } from "./util"
 import { isNotNil } from "../../../util"
 
-export const RESOURCE_SHORTCODE_REGEX = /{{< resource .*? >}}/g
```

Comment:
This is the regexp we were previously using to find `resource_link` text. The text would then get turned into shortcode objects via `Shortcode.fromString(matchingText)`.

One weakness of using regexes for this is that the regexp finds the **first** instance of `%}}` after the opening `{{% resource_link`. Ideally, it would find the **matching / closing** instance of `%}}`. So there's some danger that the match will end **too soon**. The "match ends too soon" scenario is really only a problem for legacy data, or for people who type "%}}" / ">}}% in their shortcode text. The shortcode constructor had some logic intended to catch this. Two issues:

1. It was catching slightly too many things, e.g., it would error on `{{% resource_link uuid "some text {{< sup 2 >}}" %}}` because the content included angle-shortcode delimiters.
2. We do have legacy data that triggered the error

Anyway... switching to a non-regex approach that can properly find "the matching closing delimiter" is not that hard. So that's the main change in this PR: switch to "find matching delimiter" approach.